R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+  if (expect_query_name) {
+    if (!"QUERY_NAME" %in% vars) {
+      abort(
+        paste0("Required variable `QUERY_NAME` is missing in ",
+               source_text,
+               ".")
+      )
+    }
+  }
+  if (expect_query_id) {
+    if (!"QUERY_ID" %in% vars) {
+      abort(
+        paste0("Required variable `QUERY_ID` is missing in ",
+               source_text,
+               ".")
+      )
+    }
+  }
+  if (!"TERM_NAME" %in% vars & !"TERM_ID" %in% vars) {
+    abort(
+      paste0(
+        "Variable `TERM_NAME` or `TERM_ID` is required.\n",
+        "None of them is in ",
+        source_text,
+        ".\n",
+        "Provided variables: ",
+        enumerate(vars)
+      )
+    )
+  }
+}
+
+#' Create an `smq_select` object
+#'
+#' @param name Name of the query used to select the definition of the query.
+#'
+#' @param id Identifier of the query used to select the definition of the query.
+#'
+#' @param scope Scope of the query used to select the definition of the query.
+#'
+#'   *Permitted Values*: `"BROAD"`, `"NARROW"`
+#'
+#' @details Exactly one of `name` or `id` must be specified.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [create_query_data()], [query()]
+#'
+#' @keywords source_specifications
+#'
+#' @export
+#'
+#' @return An object of class "smq_select".
+smq_select <- function(name = NULL,
+                       id = NULL,
+                       scope = NULL) {
+  out <- list(
+    name = name,
+    id = id,
+    scope = scope
+  )
+  class(out) <- c("smq_select", "list")
+  validate_smq_select(out)
+}
+
+#' Validate an object is indeed a `smq_select` object
+#'
+#' @param obj An object to be validated.
+#'
+#' @seealso [smq_select()]
+#'
+#' @author Stefan Bundfuss
+#'
+#' @export
+#'
+#' @return The original object.
+validate_smq_select <- function(obj) {
+  assert_that(inherits(obj, "smq_select"))
+  values <- unclass(obj)
+  name <- values$name
+  assert_character_scalar(name,
+                          optional = TRUE)
+  id <- values$id
+  assert_integer_scalar(id,
+                        optional = TRUE)
+  scope <- values$scope
+  assert_character_scalar(scope,
+                          values = c("BROAD", "NARROW"))
+
+  if (is.null(values$id) && is.null(values$name)) {
+    abort("Either id or name has to be non null.")
+  }
+  if (!is.null(values$id) && !is.null(values$name)) {
+    abort("Either id or name has to be null.")
+  }
+  obj
+}
+
+#' Returns a Character Representation of a `smq_select()` Object
+#'
+#' The function returns a character representation of a `smq_select()` object.
+#' It can be used for error messages for example.
+#'
+#' @param x A `smq_select()` object
+#'
+#' @param ... Not used
+#'
+#' @return A character representation of the `smq_select()` object
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [smq_select()]
+#'
+#' @keywords dev_utility
+#'
+#' @export
+#'
+#' @examples
+#'
+#' format(smq_select(id = 42, scope = "NARROW"))
+format.smq_select <- function(x, ...) {
```

Comment:
it seems this function is actually not used anywhere. Should this be removed?

Reply:
The function defines the `format` method for the `smq_select` object. It is used for error messages, e.g., in `format(queries[[i]]$definition),` it is called if `queries[[i]]$definition` is an `smq_select` object.